pkg/controller/composite/metacontroller.go

```diff
@@ -120,6 +125,19 @@ func (mc *Metacontroller) Reconcile(ctx context.Context, request reconcile.Reque
 			"[%s] Sync error - %s", cc.Name, err)
 		return reconcile.Result{}, err
 	}
+
+	// Check to see if this metacontroller instance manages the specified
+	// composite controller resource based on targeted labels.
+	if mc.targetLabelSelector != nil && !mc.targetLabelSelector.Empty() {
```

Comment:
@newtondev as an alternative, have you considered doing that in predicate ? It would skip the reconcile logic entirely  - i.e. in `server.go` line 119 and 130

Reply:
Thanks @grzesuav . I will take a look at this next.

Reply:
Added predicate filter function and passed to the watchers for both Composite and Decorator controllers. 

I will write some e2e tests for this next.